pandas/core/index.py

```diff
@@ -2745,6 +2773,44 @@ def drop_duplicates(self, keep='first'):
     def duplicated(self, keep='first'):
         return super(Index, self).duplicated(keep=keep)
 
+    def fillna(self, value=None, downcast=None):
+        """
+        Fill NA/NaN values with the specified value
+
+        Parameters
+        ----------
+        value : scalar
+            Scalar value to use to fill holes (e.g. 0).
+            This value cannot be a list-likes.
+        downcast : dict, default is None
+            a dict of item->dtype of what to downcast if possible,
+            or the string 'infer' which will try to downcast to an appropriate
+            equal type (e.g. float64 to int64 if possible)
+
+        Returns
+        -------
+        filled : Index
+        """
+        if not lib.isscalar(value):
+            msg = "'value' must be a scalar, passed: {0}"
+            raise TypeError(msg.format(type(value).__name__))
+
+        if self.hasnans:
+            return self._fillna_mask(self._isnan, value)
+        else:
+            return self._shallow_copy()
+
+    def _fillna_mask(self, mask, value, downcast=None):
```

Comment:
all of this should be in ``.putmask`` itself (which you need to change to to the ``try/except`` where the except converts to object. see #11339 this prevents having to have all of this complicaed logic